addon/link.ts

```diff
@@ -290,6 +290,10 @@ export interface UILinkParams {
   preventDefault?: boolean;
 }
 
+/**
+ * @deprecated This class will be removed in version 3 of `ember-link` in favor
+ * of only having one `Link` class
```

Comment:
```suggestion
 * of only having one {@link Link} class
```

Using [`{@link ...}`](https://tsdoc.org/pages/tags/link/) you can reference other identifiers and they will be linked in the IDE inline documentation.

Reply:
I think we should also mark the `createUILink` method on the `LinkManagerService` as `@deprecated`.

And we should consider adding a `deprecate(...)` call in the `UILink` constructor, similiar to how you did it for the `<Link>` component.